framework/tests/all-systems/Issue579.java

```diff
@@ -0,0 +1,13 @@
+// Test case for Issue579
+// https://github.com/typetools/checker-framework/issues/579
+class Issue579{
+
+    public <T>  void foo(Generic<T> real, Generic<? super T> other, boolean flag) {
+        bar(flag ? real : other);
+    }
+
+    <Q> void bar(Generic<? extends Q> param) {
+    }
+    interface Generic<F> {
```

Comment:
I would also have the empty line as separator here, like in the other test file.